sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c

```diff
@@ -248,14 +231,14 @@ int gnrc_tcp_open_active(gnrc_tcp_tcb_t *tcb,  const uint8_t address_family,
     assert(target_port != PORT_UNSPEC);
 
     /* Check AF-Family Support from target_addr */
-    switch (address_family) {
 #ifdef MODULE_GNRC_IPV6
-        case AF_INET6:
-            break;
-#endif
-        default:
-            return -EAFNOSUPPORT;
+    if (address_family != AF_INET6) {
+        return -EAFNOSUPPORT;
     }
+#else
+    return -EAFNOSUPPORT;
```

Comment:
why is that, I thought you wanted to support TCP without L3 addressing, so this should be possible?